config/config-passport.js

```diff
@@ -0,0 +1,84 @@
+import passport from 'passport';
+import passportJWT from 'passport-jwt';
+import User from '../service/schemas/users.js';
+import dotenv from 'dotenv';
+import mongoose from 'mongoose';
+
+dotenv.config();
+const secret = 'GOIT2023';
+
+const ExtractJWT = passportJWT.ExtractJwt;
+const Strategy = passportJWT.Strategy;
+const params = {
+  secretOrKey: secret,
+  jwtFromRequest: ExtractJWT.fromAuthHeaderAsBearerToken(),
+};
+
+passport.use(
+  new Strategy(params, function (payload, done) {
+    User.find({ _id: payload.id })
+      .then(([user]) => {
+        if (!user) {
+          return done(new Error('User not found'));
+        }
+        return done(null, user);
```

Comment:
in this middleware you pass user object to next middlewares. You need to narrow the user data u pass through. defenietly you should not pass user password!